### PR TITLE
[go.mod] update elastic-agent-system-metrics

### PR DESCRIPTION
**Pull request by VihasMakwana**
This PR updates `beats/v7` to point to https://github.com/elastic/beats/commit/58af1537830bf314698d760f17ba0b051f2b1dc0. 

Because of new config changes in beats and elastic-agent-system-metrics, we need to update both of them. This would also include Lee's `fbreceiver` work so it should be alright.

### CODE REVIEW

**Comment by VihasMakwana** [comment]
this is replacement for https://github.com/elastic/elastic-agent/pull/6320

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/elastic/elastic-agent
 
-go 1.22.9
+go 1.22.10
```
Comment (pkoutsovasilis) [inline go.mod:3]: I believe we shouldn't bump golang version in this PR, right? 🙂

Comment (VihasMakwana) [reply go.mod]: yeah, not needed. I've removed it.

**Review by blakerouse — APPROVED** [review]
+1

**Review by andrzej-stencel — APPROVED** [review]

**Pull request merged**